### PR TITLE
Multi select case list clearing checkbox fix

### PR DESCRIPTION
**Pull request by stephherbers**
Ticket: [USH-2262](https://dimagi-dev.atlassian.net/browse/USH-2262)

https://user-images.githubusercontent.com/36681924/181612156-1ae03346-76fe-4fad-b2be-600dbe6d5b05.mov

## Product Description

Checkboxes are not cleared when navigating between pages of results on the case list so a user can select from more than one page. However, it is still cleared if the user nagivated away from the page by either the back button or pressing "Continue"

## Technical Summary

This simply moves the code to clear the selections until the continue button is pressed (which triggers `else if (response.tree)`, rather than every time there is `response.selections` which happens with navigation. 

## Feature Flag

MULTI_SELECT_CASE_LIST

### Safety story

This is tested locally and on staging, I am also requesting QA

### Automated test coverage

None

### QA Plan

I will request QA on this simply because of visibility of this feature for BHA. This restricts the number of cases that the selection was previously cleared for, but changes like this can be sneaky. 

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js [hunk 1]
```diff
@@ -83,6 +82,9 @@ hqDefine("cloudcare/js/formplayer/menus/collections", function () {
                 _.extend(this, _.pick(response, this.detailProperties));
                 return response.details;
             } else if (response.tree) {
+                if (response.selections) {
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js:85]: This clears selections on form entry, right? That sounds good, but has some potential edge cases:

* If the user never goes into the form - they click back to the home screen, or an earlier breadcrumb - will selections get cleared?
* If the multi select case list is the first of two case lists (parent child workflow), the selections will still be present for the second case list - is that okay? (Do we even support this workflow, multi select parent and single select child? I'm pretty sure we don't support two multi select case lists.)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js [hunk 2]
```diff
@@ -64,8 +64,10 @@ hqDefine("cloudcare/js/formplayer/menus/collections", function () {
             }
             if (response.selections) {
                 urlObject.setSelections(response.selections);
-                sessionStorage.removeItem('selectedValues');
                 updateUrl = true;
+                var currentSelectedValues = sessionStorage.selectedValues !== undefined && response.multiSelect ?
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js:68]: This change seems fine, given the previous code, but it does make me curious: when is `selections` provided or not provided? I'd have thought it was provided on all `navigate_menu` responses, but that would include changing pages.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js (hunk 2 above)
Comment (stephherbers) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js:68]: ah yes I'm running into some QA bugs: [here](https://dimagi-dev.atlassian.net/browse/QA-4410) and [here](https://dimagi-dev.atlassian.net/browse/QA-4412) so not quite sure when to clear those instances and keep the pagination, but still investigating

**Comment by stephherbers** [comment]
Closing instead for https://github.com/dimagi/commcare-hq/pull/31997 
The QA bugs were blocking for this approach and it felt better to abandon this PR than try to save (sorry, PR)

**Pull request closed without merging**